File: core/src/main/java/gg/norisk/core/payloads/AbstractModuleDeactivatePayload.java
package gg.norisk.core.payloads;

import java.util.List;

/**
 * Abstract payload for deactivating specific NoRisk Client modules
 */
public abstract class AbstractModuleDeactivatePayload extends AbstractPayload {
    private final List<Modules> modules;
    
    /**
     * Creates a new module deactivate payload
     * @param modules List of modules to deactivate on the client
     */
    public AbstractModuleDeactivatePayload(List<Modules> modules) {
        super("module_deactivate");
        this.modules = modules;
    }
    
    public List<Modules> getModules() {
        return modules;
    }
} 

File: core/src/main/kotlin/gg/norisk/core/payloads/AbstractModuleDeactivatePayload.kt
package gg.norisk.core.payloads

enum class Modules {
    FovChanger,
    FullBrightModule,
    ZoomModule,
    FreeLookModule,
    NoFogModule,
    ArrowTrail,
    PackTweaks,
    ItemModel,
    AutoText,
    ItemHighlighter,
    TntTimer,
    WeatherChanger,
    TimeChanger
}

abstract class AbstractModuleDeactivatePayload(
    val modules: List<Modules>,
) : AbstractPayload("module_deactivate") {
}
